projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Companies.php

```diff
@@ -2759,8 +2761,22 @@ private function tidy_company($obj=false,$withCustomFields=false){
 
             // if have totals, pass them :)
             if (isset($obj->quotes_total)) $res['quotes_total'] = $obj->quotes_total;
-            if (isset($obj->invoices_total)) $res['invoices_total'] = $obj->invoices_total;
-            if (isset($obj->transactions_total)) $res['transactions_total'] = $obj->transactions_total;
+			if ( isset( $obj->invoices_total ) ) {
+				$company                            = $zbs->DAL->companies->getCompany( $res['id'], array( 'withInvoices' => true ) ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

Comment:
This seems unnecessarily recursive? I suspect what you're wanting is already available within `$obj`, so no need to grab `$company` anew.

Reply:
Unfortunately $obj itself doesn't include the list of invoices or transactions in order to perform calculations on them. It does include totals, which are not correct.

Reply:
Hm... I still don't like that within `getCompany()` we're calling `getCompany()`.

Currently we've got a query (actually, two depending on the params) that gets `invoices_total` (sum of all invoice values), and then we make another call to grab the entire contact including invoices, and then we overwrite `invoices_total` (hopefully in all cases).

If `invoices_total` already exists at this point (and would be "incorrect", as it includes deleted invoices), perhaps we're trying to fix it too late and should handle it earlier (e.g. in the SQL queries).

As written it works, but it's not terribly efficient. At minimum I'd swap it out for a call to `$zbs->DAL->invoices->getInvoices()`.

Reply:
> perhaps we're trying to fix it too late and should handle it earlier (e.g. in the SQL queries

Given this issue - https://github.com/Automattic/zero-bs-crm/issues/3077 - it does seem that handling it earlier, in the SQL queries, may be better. A bit like `getTransactionStatusesToIncludeQuery` but for invoices.